lib/mongos.js

```diff
@@ -317,6 +290,23 @@ class Mongos extends EventEmitter {
 
           self.state = 'stopped';
         });
+
+        // Periodically attempt to connect to server and return if successful
```

Comment:
I was thinking about this part and I wanted your opinion.

With the old logic, Promise would resolve if data was printed to stdout saying "waiting for connections". There is also an error and close handler that will reject the Promise.

With the new logic, the Promise resolves if able to connect but the error and close handler are still there to reject the Promise.

Should we reject the Promise in the error/close handler now? Rather, would it be better just to print the error but not reject the Promise and let it timeout? Not sure if this is good practice.

Reply:
I would argue that you want to keep the handlers there to reject the promise, because we _do_ want to know if some error happens or if the binary unexpectedly quits. If you have a redundant rejection it will not cause any harm.

Reply:
Okay, that's fine. Just wanted to bring the point up to discuss. Im in agreement then!